### PR TITLE
Fix bug in links_have_changed logic in message_handler

### PR DESCRIPTION
**Pull request by KyleMacPherson**
We are getting a lot of errors when we try to compared links for items in which 'expanded_links' field
is not included in their raw_json field. This PR checks for the presence of `expanded_links` before it
compares the values.

### CODE REVIEW

**Comment by KyleMacPherson** [comment]
@pmanrubia the `fetch` method did not seem to work, even when I used `with_indifferent_access` to allow querying of the hash by symbols so I went with this solution.

File: app/streams/publishing_api/message_handler.rb [hunk 1]
```diff
@@ -32,8 +32,17 @@ def is_links_update?
   end
 
   def links_have_changed?
-    current_links = old_item.raw_json['expanded_links'].deep_sort
-    new_links = new_item.raw_json['expanded_links'].deep_sort
+    current_links = if old_item && old_item.raw_json['expanded_links']
```
Comment (pmanrubia) [inline app/streams/publishing_api/message_handler.rb]: Could you please add a method in the Item to return the expanded links. You could unit test it and this method can be simplified to:

```ruby
HashDiff::Comparison.new(
  current_links.expanded_links.deep_sort,  
  new_links.expanded_links.deep_sort
).diff.present?
```

File: app/models/dimensions/item.rb [hunk 2]
```diff
@@ -30,6 +30,10 @@ def promote!(old_item)
     update(latest: true)
   end
 
+  def expanded_links
```
Comment (pmanrubia) [inline app/models/dimensions/item.rb]: Could you please add a unit test for this method? Not sure what happens when `raw_json` is nil...
